packages/cli/src/lib/decorators/preconditions/npx.spec.ts

```diff
@@ -0,0 +1,97 @@
+jest.mock('../../utils/process');
+
+import type Command from '@oclif/command';
+import * as dedent from 'dedent';
+import {constants} from 'os';
+import {mocked} from 'ts-jest/utils';
+import {spawnProcessOutput} from '../../utils/process';
+
+import {IsNpxInstalled} from './npx';
+
+describe('IsNpxInstalled', () => {
+  const mockedSpawnProcessOutput = mocked(spawnProcessOutput);
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('when npx is not installed', () => {
+    beforeEach(() => {
+      mockedSpawnProcessOutput.mockResolvedValue({
+        exitCode: constants.errno.ENOENT,
+        stderr: '',
+        stdout: '',
+      });
+    });
+
+    it('should return false and warn', async () => {
+      const fakeCommand = {
+        id: 'foo',
+        warn: jest.fn(),
+      };
+
+      await expect(
+        IsNpxInstalled()((fakeCommand as unknown) as Command)
+      ).resolves.toBe(false);
+      expect(fakeCommand.warn).toHaveBeenCalledTimes(2);
+      expect(fakeCommand.warn).toHaveBeenCalledWith(dedent`
+      foo requires npx to run.
+      Newer version Node.js comes bundled with npx.
+    `);
+      expect(fakeCommand.warn).toHaveBeenCalledWith(dedent`
+       Please visit https://github.com/coveo/cli/wiki/Node.js,-NPM-and-NPX-requirements for more detailed installation information.
+      `);
+    });
+  });
+
+  describe('when an unknown error happens while checking for npx', () => {
+    beforeEach(() => {
+      mockedSpawnProcessOutput.mockResolvedValue({
+        exitCode: 1,
+        stderr: 'some random error oh no',
+        stdout: '',
+      });
+    });
+
+    it('should return false and warn', async () => {
+      const fakeCommand = {
+        id: 'foo',
+        warn: jest.fn(),
+      };
+
+      await expect(
+        IsNpxInstalled()((fakeCommand as unknown) as Command)
+      ).resolves.toBe(false);
+      expect(fakeCommand.warn).toHaveBeenCalledTimes(2);
+      expect(fakeCommand.warn).toHaveBeenCalledWith(dedent`
```

Comment:
> I'm not sure I would have a test that verifies the "exact same message contained in the command, just copy pasted".
>
> Whenever we want to change a comma for the warning message, the tests will start to fail for no "good" reason.
>
> I would focus on regex matching a couple important part: the readme link, a "high level check" on some important part of the message (eg: /some random error oh no/ or /a valid npx installation/, stuff like this).
>
> The exact wording itself does not seem very important to validate with a UT.

@olamothe , https://github.com/coveo/cli/pull/121#discussion_r604182036

That would be true for end-to-end tests, but it is not for unit test imo.
Unit tests IMO must be as 'strict' as possible.

> Whenever we want to change a comma for the warning message, the tests will start to fail for no "good" reason.
That's technically a change of behavior, and it must be tested, as all other are. Do you prefer an automated test, or that a QA specialist spend time checking it? I for sure know that I'd prefer the former.